javascript/ql/src/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -209,10 +209,49 @@ private module Mongoose {
     result = getAMongooseInstance().getAMemberCall("createConnection")
   }
 
+  /**
+   * A common interface for various Mongoose calls.
+   */
+  private class CommonInterface extends DataFlow::InvokeNode {
+    /**
+     * Holds if this call returns an object of type `Query`.
+     */
+    abstract predicate returnsQuery();
+
+    /**
+     * Holds if this call returns a `Query` that evaluates to a one or
+     * more Documents (`asArray` is false if it evaluates to a single
+     * Document).
+     */
+    abstract predicate returnsDocumentQuery(boolean asArray);
+
+    /**
+     * Holds if this call interprets `arg` as a query.
+     */
+    abstract predicate interpretsArgumentAsQuery(DataFlow::Node arg);
+  }
+
   /**
    * Provides classes modeling the Mongoose Model class
    */
   module Model {
+    private class ModelAsCommonInterface extends CommonInterface, DataFlow::MethodCallNode {
```

Comment:
This could be a `MongooseModelMethodCall` (or `ModelMethodCall`). 

(And likewise for the other classes)